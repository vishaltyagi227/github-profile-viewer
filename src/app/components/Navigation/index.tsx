import React from 'react';
import { NavLink } from 'react-router-dom';

import { ROUTE, PROTECTED_ROUTE } from '@constant/index';
import INavbar from '@containers/Navbar/type';
import Link from '@components/Link';
import Button from '@components/Button';
import useStyles from './style';

const Navigation = ({ isLogged, handleClick }: INavbar) => {
  const classes = useStyles();

  return (
    <nav>
      <ul className={classes.navList}>
        <li className={classes.navItem}>
          <Link
            btn
            component={NavLink}
            to={ROUTE.Home}
            activeClassName={classes.active}
            exact
            className={classes.navLink}
          >
            Search
          </Link>
        </li>
        {isLogged && (
          <>
            <li className={classes.navItem}>
              <Link
                btn
                component={NavLink}
                to={PROTECTED_ROUTE.Follow}
                activeClassName={classes.active}
                exact
                className={classes.navLink}
              >
                Who to Follow
              </Link>
            </li>
            <li className={classes.navItem}>
              <Link
                btn
                component={NavLink}
                to={PROTECTED_ROUTE.Profile}
                activeClassName={classes.active}
                exact
                className={classes.navLink}
              >
                Profile
              </Link>
            </li>
          </>
        )}
        {isLogged ? (
          <>
            <li className={classes.navItem}>
              <Button className={classes.navLink} onClick={handleClick}>
                Logout
              </Button>
            </li>
          </>
        ) : (
          <>
            <li className={classes.navItem}>
              <Link
                btn
                component={NavLink}
                to={ROUTE.Login}
                activeClassName={classes.active}
                exact
                className={classes.navLink}
              >
                Login
              </Link>
            </li>
          </>
        )}
      </ul>
    </nav>
  );
};

export default Navigation;
