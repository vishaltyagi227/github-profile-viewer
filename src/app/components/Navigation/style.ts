import { Theme } from '@mui/material/styles';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navList: {
      display: 'flex',
    },
    navItem: {
      listStyle: 'none',
      marginLeft: theme.spacing(2),
    },
    navLink: {
      color: 'inherit',
      '&:hover': {
        backgroundColor: theme.customPallete.primary.light,
      },
    },
    active: {
      backgroundColor: theme.customPallete.primary.light,
    },
  })
);

export default useStyles;
