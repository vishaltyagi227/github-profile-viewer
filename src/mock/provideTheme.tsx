import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
