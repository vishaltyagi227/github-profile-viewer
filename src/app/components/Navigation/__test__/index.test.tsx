import React from 'react';
import { render, RenderResult, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import { BrowserRouter } from 'react-router-dom';
import { ROUTE, PROTECTED_ROUTE } from '@constant/index';
import Navigation from '../index';

describe('<Navigation /> when user not login', () => {
  let container: RenderResult;
  beforeEach(() => {
    container = render(
      provideTheme(
        <BrowserRouter>
          <Navigation isLogged={false} handleClick={jest.fn()} />
        </BrowserRouter>
      )
    );
  });

  afterEach(cleanup);

  it('Should render Correctly without rendering the protected link ', () => {
    expect(container.getByText('Search')).toBeInTheDocument();
    expect(container.getByText('Login')).toBeInTheDocument();
    expect(container.queryByText('Who to Follow')).not.toBeInTheDocument();
    expect(container.queryByText('Logout')).not.toBeInTheDocument();
  });

  it('Should have correct route url', () => {
    expect(container.getByText('Search')).toHaveAttribute('href', ROUTE.Home);
    expect(container.getByText('Login')).toHaveAttribute('href', ROUTE.Login);
  });
});

describe('<Navigation /> when user Login', () => {
  let container: RenderResult;
  beforeEach(() => {
    container = render(
      provideTheme(
        <BrowserRouter>
          <Navigation isLogged handleClick={jest.fn()} />
        </BrowserRouter>
      )
    );
  });

  afterEach(cleanup);

  it('Should render correctly with rendering the protected link ', () => {
    expect(container.getByText('Search')).toBeInTheDocument();
    expect(container.queryByText('Login')).not.toBeInTheDocument();
    expect(container.getByText('Who to Follow')).toBeInTheDocument();
    expect(container.getByText('Profile')).toBeInTheDocument();
    expect(container.getByText('Logout')).toBeInTheDocument();
  });

  it('Should have correct route url', () => {
    expect(container.getByText('Search')).toHaveAttribute('href', ROUTE.Home);
    expect(container.getByText('Who to Follow')).toHaveAttribute(
      'href',
      PROTECTED_ROUTE.Follow
    );
    expect(container.getByText('Profile')).toHaveAttribute(
      'href',
      PROTECTED_ROUTE.Profile
    );
    expect(container.getByText('Logout')).toHaveAttribute('type', 'button');
  });
});

describe('<Navigation /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with snapshot when user not login', () => {
    const tree = renderer
      .create(
        provideTheme(
          <BrowserRouter>
            <Navigation isLogged={false} handleClick={jest.fn} />
          </BrowserRouter>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match with snapshot when user logged', () => {
    const tree = renderer
      .create(
        provideTheme(
          <BrowserRouter>
            <Navigation isLogged handleClick={jest.fn} />
          </BrowserRouter>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
