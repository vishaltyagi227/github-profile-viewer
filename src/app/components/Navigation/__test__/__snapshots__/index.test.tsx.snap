// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snapshot Should match with snapshot when user logged 1`] = `
<nav>
  <ul
    className="makeStyles-navList-19"
  >
    <li
      className="makeStyles-navItem-20"
    >
      <a
        aria-current="page"
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root makeStyles-active-22"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
        tabIndex={0}
      >
        Search
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </li>
    <li
      className="makeStyles-navItem-20"
    >
      <a
        aria-current={null}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        href="/follow"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Who to Follow
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </li>
    <li
      className="makeStyles-navItem-20"
    >
      <a
        aria-current={null}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        href="/profile"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Profile
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </li>
    <li
      className="makeStyles-navItem-20"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Logout
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </li>
  </ul>
</nav>
`;

exports[`<Navigation /> Snapshot Should match with snapshot when user not login 1`] = `
<nav>
  <ul
    className="makeStyles-navList-19"
  >
    <li
      className="makeStyles-navItem-20"
    >
      <a
        aria-current="page"
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root makeStyles-active-22"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
        tabIndex={0}
      >
        Search
      </a>
    </li>
    <li
      className="makeStyles-navItem-20"
    >
      <a
        aria-current={null}
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-navLink-21 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        href="/login"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;
